### PR TITLE
Feature/nonconst set visit

### PR DESCRIPTION
**Pull request by joaquintides**
Addresses #260.

### CODE REVIEW

**Review by pdimov — APPROVED** [review]

File: test/cfoa/visit_tests.cpp [hunk 1]
```diff
@@ -971,6 +976,158 @@ namespace {
...
+
+    auto bound_f = [&] {
+      ++num_started;
+      f([&] (const mutable_pair& x) {
+        ++in_visit;
+        ++x.second;
+
+        {
+          std::unique_lock<std::mutex> lk(m);
+          cv.wait(lk, [&] { return finish; });
```
Comment (cmazakas) [inline test/cfoa/visit_tests.cpp:1045]: I think we should have a latch component either internally or in Compat we can use here. I'm assuming the intention here was to block all the spawned threads until we're ready for them to actually run, right?

File: test/cfoa/visit_tests.cpp [hunk 2]
```diff
@@ -971,6 +976,158 @@ namespace {
...
+  };
+
+  struct null_mutable_pair
+  {
+    operator mutable_pair() const { return {0,0}; }
+  };
+
+  struct mutable_pair_hash
+  {
+    using is_transparent = void;
```
[the remark this answers is not in the document]
Comment (joaquintides) [reply test/cfoa/visit_tests.cpp:994]: It is `is_avalanching` where we've made this change. As for `is_transparent`, its behavior is dictated by the standard, and we're using `using is_transparent = void` elsewhere in the tests.

File: test/cfoa/visit_tests.cpp (hunk 1 above)
Comment (joaquintides) [reply test/cfoa/visit_tests.cpp]: You're absolutely right! Changed it to use `boost::compat::latch`.

File: test/cfoa/visit_tests.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmazakas) [reply test/cfoa/visit_tests.cpp:994]: Ah, thanks. Got my wires crossed there.

**Review by cmazakas — APPROVED** [review]

**Pull request merged**